### DISCUSSION TITLE
The TIDAL desktop app is reporting incomplete media metadata to Windows

### DISCUSSION THREAD

**Discussion by ungive** [tidal-music/discussions#74] [Report bug]
I have been redirected here by TIDAL support. My original message:

TIDAL for Windows reports music metadata (title, artist, album, playback position, etc.) to the operating system, so that other applications can see what the TIDAL media player is currently playing. I am using a low-level C++ API, called WinRT, to access that information. When I attempt to read the album name it reports no information at all.

The following low-level C++ functions (of the WinRT API) are used to retrieve this information:
Album: `Windows.Media.Control.GlobalSystemMediaTransportControlsSessionMediaProperties.AlbumTitle()`
Documentation: https://learn.microsoft.com/en-us/uwp/api/windows.media.control.globalsystemmediatransportcontrolssessionmediaproperties?view=winrt-22621

The return value of `AlbumTitle()` is empty:

Screenshot 1 (s1) contains the code that is used to retrieve the information:

![s1](https://github.com/tidal-music/discussions/assets/24623262/33ee1405-6d28-49a1-86ba-6a87a72e012a)

session is an instance of the C++ type `winrt::Windows::Media::Control::GlobalSystemMediaTransportControlsSession`.

Screenshot 2 (s2) shows the contents of the variables of screenshot 1:

![s2](https://github.com/tidal-music/discussions/assets/24623262/50c7efee-e4a8-44cd-80c5-abea31832c48)

As you can see the album title is empty. This information is useful and should be reported by TIDAL, so that it's possible to more distinctly distinguish which song exactly is playing (e.g. to retrieve the correct cover image or to know it's playing from the a single album or from a normal album or from a deluxe version of the album).

It would be great if TIDAL could report more detailed information via system APIs in the future!

Thank you.

---

Here is also a link to a GitHub issue outlining the problem further: https://github.com/ungive/discord-music-presence/issues/17

I hope someone here can be of help and perhaps forward the issue to the right department.

Thank you!

**Comment by stefanmoro** [comment]
Hi @ungive ,

This is strictly speaking a question to customer support since it refers to an existing product offering (desktop app). However, I have forwarded the info to the devs working on our desktop app so they are aware. Can't make any promises about if/when it'll be fixed though.

**Comment by enjikaka** [comment]
We don't have any special handling for this in our Electron based desktop applications. We're merely using the [MediaSession API](https://developer.mozilla.org/en-US/docs/Web/API/MediaSession) from client side JavaScript.

If the album title is missing in the lower level Windows APIs, that'd be an issue for Chromium. I could find a [similar ticket for macOS](https://issues.chromium.org/issues/40889068) but not Windows.

**Reply by ungive** [reply]
I have reported the issue here: https://issues.chromium.org/issues/349310439

**Reply by enjikaka** [reply]
Seems they fixed the issue on macOS, so lets hope they manage on Windows too. :)
